core/domain/exp_jobs_one_off.py

```diff
@@ -528,6 +528,120 @@ def reduce(key, values):
         yield (key, values)
 
 
+class ExplorationMathRichTextInfoModelGenerationOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that finds all the explorations with math rich text components and
+    creates a temporary storage model with all the information required for
+    generating math rich text component SVG images.
+    """
+
+    _SUCCESS_KEY = 'exploration-with-math-tags'
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        try:
+            exploration.validate()
+        except Exception as e:
+            logging.error(
+                'Exploration %s failed non-strict validation: %s' %
+                (item.id, e))
+            yield (
+                'validation_error',
+                'Exploration %s failed non-strict validation: %s' %
+                (item.id, e))
+            return
+        html_strings_in_exploration = ''
+        for state in exploration.states.values():
+            html_strings_in_exploration += (
+                ''.join(state.get_all_html_content_strings()))
+        list_of_latex_values_without_svgs = (
+            html_validation_service.
+            get_latext_values_without_svg_from_html(
+                html_strings_in_exploration))
+        if len(list_of_latex_values_without_svgs) > 0:
+            yield (
+                ExplorationMathRichTextInfoModelGenerationOneOffJob.
+                _SUCCESS_KEY,
+                (item.id, list_of_latex_values_without_svgs))
+
+    @staticmethod
+    def reduce(key, values):
+        if key == (
+                ExplorationMathRichTextInfoModelGenerationOneOffJob.
+                _SUCCESS_KEY):
+            final_values = [ast.literal_eval(value) for value in values]
+            estimated_no_of_batches = 1
+            approx_size_of_math_svgs_bytes_in_a_batch = 0
+            multiple_explorations_math_rich_text_info = []
+            longest_raw_latex_string = ''
+            total_number_of_svgs_required = 0
+            for value in final_values:
+                math_rich_text_info = (
+                    exp_domain.ExplorationMathRichTextInfo(value[1]))
+                multiple_explorations_math_rich_text_info.append(
+                    (math_rich_text_info, value[0]))
+
+                approx_size_of_math_svgs_bytes = (
+                    math_rich_text_info.get_svg_size_in_bytes())
+                longest_raw_latex_string_in_exploration = (
```

Comment:
If this variable is not used multiple times then should we directly use `math_rich_text_info.get_largest_latex_value()` below?

Reply:
No, We need to find math_rich_text_info.get_longest_latex_value() for each exploration and then find the longest  of them

Reply:
Can we do this on line 602:

```
longest_raw_latex_string = max(
                        math_rich_text_info.get_longest_latex_value(),
                        longest_raw_latex_string, key=len)

```

(just to avoid loading lots of variables in the brain while reading this function!)

Reply:
done